### PR TITLE
Connect bookstore app to API

### PR DESCRIPTION
**Pull request by Fene-87**
In this project milestone, I connect my bookstore application to an existing API to create and remove books from a server.
- I use axios as well as createAsyncThunk to make API calls.
- I modify reducer logic to include HTTP requests to the Bookstore API.

The application functions just as before.

### CODE REVIEW

File: src/components/book/Book.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
-import { removeBook } from '../../redux/features/books/booksSlice';
+import { deleteBooks } from '../../redux/features/books/booksSlice';
 import './Book.css';
 
 const Book = ({
-  itemId,
+  id,
   title,
   author,
   category,
```
Comment (huseyinbiyik) [inline src/components/book/Book.js:9]: You did a brilliant job for sure. Add and remove functions working properly 👍  But let's use useEffect for fetching the books as requested on project requirements.
Screenshot 2023-03-02 at 7 34 21 PM

**Review by huseyinbiyik — CHANGES_REQUESTED** [review]
Hi @Fene-87,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
Almost there

## Highlights
- You used axios and error handling looks great 👍 
- Linter set up well and checks pass 🟢
- Git flow followed 👌
- .gitignore used to ignore requested files 📁
- Commit messages are meaningful and written in imperative tense 📝
- PR description is descriptive 📝

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/book/Book.js (hunk 1 above)
Comment (Fene-87) [reply src/components/book/Book.js:9]: Hey @huseyinbiyik Thank you for your feedback I appreciate. You have pointed out that I should fetch using the useEffect. That is exactly what I did. If you go to my pages/books, which is where I passed in the Book component, you'll notice that I used useEffect to fetch the books from the API.
HuseyinCR

It is after fetching that I pass the data to the book component as props. The point is that I fetch books from the bookstore API when the component has rendered using useEffect. Kindly take that into consideration.

Thank you.

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢 🟢 🟢

Hi @Fene-87,

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:

✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ Used redux createAsyncThunk hook and react's useEffect hook to dispatch actions 👍🏽
✅ Adding and removing books functionalities work fine as expected 👍🏽
✅ The app runs fine as expected 👍🏽

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by Fene-87** [comment]
Hey @AmaduKamara Thank you so much for your time and feedback

File: src/components/book/Book.js (hunk 1 above)
Comment (huseyinbiyik) [reply src/components/book/Book.js:9]: @Fene-87 ops, I am so sorry that I didn't notice that. Sometimes, it is happening. But one thing I am sure of is your code looks brilliant in general 👍

**Pull request merged**